manifests/config.pp

```diff
@@ -27,9 +27,15 @@
 #     ensure => 'blank',
 #   }
 #
+#   # By value type/size:
+#   postfix::config {
+#     'myorigin':      value => undef  # Rm
+#     'myhostname':    value => 'xyz'  # Set 
+#     'mydestination': value => '',    # Clear
+#   }
 define postfix::config (
-  Optional[String]                   $value  = undef,
-  Enum['present', 'absent', 'blank'] $ensure = 'present',
+  Optional[String]                             $value  = undef,
+  Optional[Enum['present', 'absent', 'blank']] $ensure = undef,
```

Comment:
There is no need for `ensure` to be optional. Please leave the parameters as they were before.

Reply:
Do you mean 'as they were before' is `Variant` or just `Enum['present', 'absent', 'blank']`?
If there is no option for undef than the idea to remove entry from configuration is missing.

Reply:
Well removal is taken care of with `ensure => absent`. Actually, I thought for a while that this PR added the support for `clear` but it already exists with `ensure => blank`, so now I'm not sure what is the point at all...

Reply:
This PR is to extend functionality, leaving old in place. Tried to describe with [comment](https://github.com/camptocamp/puppet-postfix/pull/290#issue-541196745) and [comment](https://github.com/camptocamp/puppet-postfix/pull/290#discussion_r558073909). The idea is to be able to create profiles when you can pass all config from hiera. The easiest way just to check value type, other approaches add  too much complexity into profile or hiera.